dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -0,0 +1,340 @@
+                if not os.path.exists(directory):
+                    os.makedirs(directory)
+
+                # last acquisition
+                if total_run_time > duration-max_duration:
+                    duration = duration-total_run_time
+                    logger.info('Going to tell psyllid to start the last run')
+                    payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':duration}
+                    try:
+                        self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
+                    except core.exceptions.DriplineError:
+                        logger.error('Error from psyllid provider or psyllid')
+                        payload = {'channel': self.channel_id}
+                        try:
+                            self.provider.cmd(self.daq_target, 'unblock_channel', payload = payload)
+                        finally:
+                            raise core.exceptions.DriplineGenericDAQError('Starting psyllid run failed')
+                    except Exception as e:
+                        logger.error('Local error')
+                        payload = {'channel': self.channel_id}
+                        try:
+                            self.provider.cmd(self.daq_target, 'unblock_channel', payload=payload)
+                        finally:
+                            raise e
+
+                # acquisitions with duration = max_duration
+                else:
+                    total_run_time+=max_duration
+                    logger.info('Going to tell psyllid to start run')
+                    payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':max_duration}
+
+                    try:
+                        self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
+                        time.sleep(self._max_duration)
+                    except core.exceptions.DriplineError:
+                        logger.error('Error from psyllid provider or psyllid')
+                        payload = {'channel': self.channel_id}
+                        try:
+                            self.provider.cmd(self.daq_target, 'unblock_channel', payload = payload)
+                        finally:
+                            raise core.exceptions.DriplineGenericDAQError('Starting psyllid run failed')
+                    except Exception as e:
+                        logger.error('Local error')
+                        payload = {'channel': self.channel_id}
+                        try:
+                            self.provider.cmd(self.daq_target, 'unblock_channel', payload=payload)
+                        finally:
+                            raise e
+
+
+
+    def _stop_data_taking(self):
+        if self.is_running:
+            self.provider.cmd(self.psyllid_interface, 'stop_run', payload = self.payload_channel)
```

Comment:
there is already an end_run method in daq provider. This method calls the _stop_data_taking method of the daq target.